src/services/review/RepoCheckerService.ts

```diff
@@ -0,0 +1,235 @@
+        ).andThen((setOfAllPermalinks) =>
+          ok(new Set([...setOfAllDocumentsPath, ...setOfAllPermalinks]))
+        )
+      })
+      .andThen((setOfAllDocumentsAndPagesPath) => {
+        mdFiles.forEach(async (fileName) => {
+          const file = fs.readFileSync(path.join(repoPath, fileName), "utf8")
+          const filePermalink = file
+            .split("---")[1]
+            .split("permalink: ")[1]
+            .split("\n")[0]
+          const fileContent = file.split("---")[2]
+          if (!fileContent) {
+            return
+          }
+          const html = await marked.parse(fileContent)
+          const dom = new JSDOM(html)
+          const anchorTags = dom.window.document.querySelectorAll("a")
+          forEach(anchorTags, (tag) => {
+            const href = tag.getAttribute("href")
+            const text = tag.textContent
+            if (href) {
+              console.log(href)
+              const encodedHref = this.documentPathDecoder(href)
+              if (
+                !this.isExternalLinkOrPageRef(encodedHref) &&
+                !setOfAllDocumentsAndPagesPath.has(encodedHref) &&
+                // Amplify supports adding of trailing slash
+                !setOfAllDocumentsAndPagesPath.has(`${encodedHref}/`)
+              ) {
+                console.log(setOfAllDocumentsAndPagesPath)
+                this.persistLogs(repo, [
+                  new SiteCheckerError(
+                    `Broken link ${href} that links ${text} found in ${fileName} which has permalink ${filePermalink}`
+                  ),
+                ])
+              }
+            }
+          })
+          const imgTags = dom.window.document.querySelectorAll("img")
+          forEach(imgTags, (tag) => {
+            const src = tag.getAttribute("src")
+            if (src) {
+              const encodedSrc = this.documentPathDecoder(src)
+              if (
+                !this.isExternalLinkOrPageRef(encodedSrc) &&
+                !setOfAllDocumentsAndPagesPath.has(encodedSrc)
+              ) {
+                this.persistLogs(repo, [
+                  new SiteCheckerError(
+                    `Broken image ${src} found in ${fileName} which has permalink ${filePermalink}`
+                  ),
+                ])
+              }
+            }
+          })
+        })
+        console.log("Site checker done")
+        return ok(undefined)
+      })
+    try {
+      console.log(await checker)
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  isExternalLinkOrPageRef = (link: string) =>
+    link.startsWith("http") ||
+    link.startsWith("www") ||
+    link.startsWith("mailto:") ||
+    link.startsWith("#")
+
+  documentPathDecoder = (documentPath: string) => {
+    for (let i = 0; i < ALLOWED_FILE_EXTENSIONS.length; i += 1) {
+      if (documentPath.endsWith(ALLOWED_FILE_EXTENSIONS[i])) {
+        return decodeURIComponent(documentPath)
+      }
+    }
+    return documentPath
+  }
```

Comment:
ill do this in the upstream prs instead when i make all the util functions use neverthrow
also wait both of these use anon functions, what do you mean by standardisation ya?

Reply:
It was more like sometimes it is:

```ts
function() {
    // definition
}
```

And other times it is (Iike here):

```ts
function = () => {
    // definition
}
```

I think it would be better if we standardised the way the functions are defined, just a minor code style nit.

Reply:
eh wth this was a brain fart, will convert heh

Reply:
wah sorry this was a brain fart moment

62d18f4a